src/Controller/AdminController.php

```diff
@@ -3,9 +3,23 @@
 namespace App\Controller;
 
 use App\Model\UserManager;
+use App\Model\GenieManager;
+use App\Model\SpecialtyManager;
 
 class AdminController extends AbstractController
 {
+    private UserManager $userManager;
+    private GenieManager $genieManager;
+    private SpecialtyManager $specialtyManager;
+
+    public function __construct()
+    {
+        parent::__construct();
+        $this->userManager = new UserManager();
+        $this->genieManager = new GenieManager();
+        $this->specialtyManager = new SpecialtyManager();
```

Comment:
SpecialtyManager doesn't seem to be used anywhere, so why do you have it in your controller ?

Reply:
specialtyManager is used in the function showUpdateGenie